src/gt4py/storage/allocators.py

```diff
@@ -145,10 +167,10 @@ def allocate(
         self,
         shape: Sequence[core_defs.IntegralScalar],
         dtype: core_defs.DType[_ScalarT],
-        layout_map: Sequence[int],
+        layout_map: BufferLayoutMap,
         device: core_defs.Device,
-        byte_alignment: Optional[int] = None,
-        aligned_index: Sequence[int] = None,
+        byte_alignment: int,
```

Comment:
I made this required, for the lazy reason that the current allocator cannot handle the case where it's `None`. If we want a default, would `byte_alignment=1` be good enough? Or would `byte_alignment=None` mean get a good default for the device? (probably not, because that shouldn't be a concern of the low level allocator?)